src/topology/algebra/ordered.lean

```diff
@@ -2178,22 +2179,140 @@ lemma tendsto_of_monotone {ι α : Type*} [preorder ι] [topological_space α]
 if H : bdd_above (range f) then or.inr ⟨_, tendsto_at_top_csupr h_mono H⟩
 else or.inl $ tendsto_at_top_at_top_of_monotone' h_mono H
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Ioi a)) (nhds_within (a⁻¹) (Iio (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Iio a)) (nhds_within (a⁻¹) (Ioi (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Ioi (a⁻¹))) (nhds_within a (Iio a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Ioi _ _ _ _ (a⁻¹)
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Iio (a⁻¹))) (nhds_within a (Ioi a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Iio _ _ _ _ (a⁻¹)
+
+lemma nhds_cleft_sup_nhds_cright (a : α) [topological_space α] [linear_order α] :
+  nhds_within a (Iic a) ⊔ nhds_within a (Ici a) = 𝓝 a :=
+by rw [← nhds_within_union, Iic_union_Ici, nhds_within_univ]
+
+lemma nhds_oleft_sup_nhds_cright (a : α) [topological_space α] [linear_order α] :
+  nhds_within a (Iio a) ⊔ nhds_within a (Ici a) = 𝓝 a :=
+by rw [← nhds_within_union, Iio_union_Ici, nhds_within_univ]
+
+lemma nhds_cleft_sup_nhds_oright (a : α) [topological_space α] [linear_order α] :
+  nhds_within a (Iic a) ⊔ nhds_within a (Ioi a) = 𝓝 a :=
+by rw [← nhds_within_union, Iic_union_Ioi, nhds_within_univ]
+
+lemma continuous_at_iff_continuous_cleft_cright [topological_space α] [linear_order α]
+  [topological_space β] {a : α} {f : α → β} :
+  continuous_at f a ↔ continuous_within_at f (Iic a) a ∧ continuous_within_at f (Ici a) a :=
+by simp only [continuous_within_at, continuous_at, ← tendsto_sup, nhds_cleft_sup_nhds_cright]
+
+lemma continuous_on_Icc_extend_from_Ioo [topological_space α] [linear_order α] [densely_ordered α]
+  [order_topology α] [topological_space β] [t2_space β] [regular_space β] {f : α → β} {a b : α}
```

Comment:
You shouldn't need `t2_space` here, `regular_space` is a stronger condition (and I think the corresponding instance exists).